src/main/java/dev/waterdog/waterdogpe/packs/PackManager.java

```diff
@@ -150,23 +150,27 @@ public void rebuildPackets() {
 
         this.stackPacket.getBehaviorPacks().clear();
         this.stackPacket.getResourcePacks().clear();
+
         this.stackPacket.setGameVersion("");
 
         for (ResourcePack pack : this.packs.values()) {
-            if (!pack.getType().equals(ResourcePack.TYPE_RESOURCES)) {
-                continue;
-            }
             ResourcePacksInfoPacket.Entry infoEntry = new ResourcePacksInfoPacket.Entry(pack.getPackId().toString(), pack.getVersion().toString(),
                     pack.getPackSize(), "", "", "", false, false);
-            this.packsInfoPacket.getResourcePackInfos().add(infoEntry);
             ResourcePackStackPacket.Entry stackEntry = new ResourcePackStackPacket.Entry(pack.getPackId().toString(), pack.getVersion().toString(), "");
-            this.stackPacket.getResourcePacks().add(stackEntry);
+            if (pack.getType().equals(ResourcePack.TYPE_RESOURCES)) {
+                this.packsInfoPacket.getResourcePackInfos().add(infoEntry);
+                this.stackPacket.getResourcePacks().add(stackEntry);
+                continue;
```

Comment:
I think you should use else if here instead of a second if branch, and also the continue; seems unnessecaryto me

Reply:
No problem, I used "continue" to avoid using "else if".